libs/scripts/scripts.js

```diff
@@ -19,6 +19,14 @@ import {
 // Production Domain
 const prodDomains = ['milo.adobe.com'];
 
+const stageDomainsMap = {
```

Comment:
Would we need the same mapping to be configured by each consumer or is it enough to have it in Milo? Should consumers only add new values or maybe contribute the mapping to Milo directly ?

Reply:
The idea of the current solution is to be agnostic. Each consumer would need to add their own mapping and, in this way, will have complete control over which links get converted to their stage counterparts. Milo would not have any default mapping that the consumer can add to.

Reply:
This is in `scripts.js`, thus only specific to Milo to make it easy for people previewing things inside Milo's Sharepoint to test this. It will be up to individual consumers to define which hostnames they want converted to their stage counterparts. I'm not sure whether Milo should maintain a list of all its consumer's hostnames, that may be overhead we'd rather avoid.

I also get your concern that multiple consumers would have to maintain a similar map, but if we think of Milo as a sort of 3rd party, it would never have client-specific information within it, instead it would ask client to define setting through a config.

Maintaining such a list in a spreadsheet is also not ideal, since it would entail an extra request; granted, it would be just for Stage, but it would still affect performance in a critical point of the page's loading timeline.

Do you have any other ideas of how we could achieve this without overhead in either Milo or an extra network request?

Reply:
Great feedback, thank you. Indeed, my problem is with repetitive content (and more specific global navigation content where we have all sort of links), but it makes sense to ask consumers to define their own mappings if we're trying to be agnostic. There might be consumers that don't need the navigation or these mappings.
As long as we're able to define a full mapping in `federal`, I think most of the limitations should be automatically solved.